ros/processor/suggestions_engine.py

```diff
@@ -64,25 +93,125 @@ def run(self):
             self.consumer.close()
 
 
-def download_and_extract(service, event, archive_URL, host, org_id):
-    logging.info(f"{service} - {event} - Downloading the report for system {host.get('id')}.")
+def find_root_directory(directory, target_file):
+    """
+    Recursively search for the target file in the given directory - to get root directory of an archive.
+    """
+    for dirpath, dirnames, filenames in os.walk(directory):
+        if target_file in filenames:
+            return dirpath
+
+    return None
+
+
+def get_index_file_path(service, event, host, extracted_dir):
+    extracted_dir_root = find_root_directory(extracted_dir, "insights_archive.txt")
+
+    if not extracted_dir_root:
+        logging.error(
+            f"{service} - {event} - insights_archive.txt not found in the extracted dir for system {host.get('id')}."
+        )
+        return None
+
+    pmlogger_dir = os.path.join(extracted_dir_root, "data/var/log/pcp/pmlogger/")
+
+    index_files = [
+        file for file in os.listdir(pmlogger_dir) if file.endswith(".index")
+    ]
+    index_file_path = os.path.join(pmlogger_dir, index_files[0])
+
+    return index_file_path
+
 
-    response = requests.get(archive_URL, timeout=10)
+def create_output_dir(request_id, host):
+    output_dir = f"/tmp/pmlogextract-output-{request_id}/"
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
 
-    if response.status_code != HTTPStatus.OK:
+    logging.debug(f"Successfully created output_dir for system {host.get('id')}: {output_dir}")
+
+    return output_dir
+
+
+def run_pcp_commands(service, event, host, index_file_path, request_id):
+    try:
+        sanitized_request_id = request_id.replace("/", "_")
+        output_dir = create_output_dir(sanitized_request_id, host)
+
+        pmlogextract_command = [
+            "pmlogextract",
+            "-c",
+            "ros/lib/pcp_extract_config",
+            index_file_path,
+            output_dir
+        ]
+
+        logging.info(f"{service} - {event} - Running pmlogextract command for system {host.get('id')}.")
+        try:
+            subprocess.run(pmlogextract_command, check=True)
+            logging.info(f"{service} - {event} - Successfully ran pmlogextract command for system {host.get('id')}.")
+        except subprocess.CalledProcessError as error:
+            logging.error(
+                f"{service} - {event} - Error running pmlogextract command for system {host.get('id')}: {error.stdout}"
+            )
+            return
```

Comment:
Just thought we can have separate methods for `pmlogextract` and `pmlogsummary`?